### DISCUSSION TITLE
QR code not rendering on mobile (android build)

### DISCUSSION THREAD

**Discussion by Dilosen** [WalletConnect/WalletConnectUnity#142] [Q&A]
Hi All 

I only tested the sample scene with the exisiting modal. I have added two extra chains (algorand and algoTestNet) respectively.  I have then built the scene out onto android to test it's interface and how it works. (my games are on android devices)

Everything pulls fine from the walletConnect Explorer such as names, icons etc, but the QR code to access the wallet (or a button to link the installed wallet app) does not appear. I assume this is handled by and internal script and not walletconnect?

Assumption: Click on modal -> click on wallet -> click on "connect" button (not qr) -> open wallet/dapp

This does not seem to be the case as we can only get as far as selecting a wallet from the wallet list

**Comment by skibitsky** [comment]
Hey,

On mobile, the flow is slightly different than on desktop for better UX.

When you select a wallet from the list on mobile, it will try to open the wallet app with a deep link. The deep link is loaded from the explorer. It's still possible to open a QR code by going to the `All Wallets` and tapping on the `QR Code` button on the right side of the search bar. There will also be a `Copy Link` button, which copies the linking URI that can be manually pasted into some wallets.

**Reply by Dilosen** [reply]
okay I see now as I tested it again. I'm assuming all of that is handled on the back-end right? 

When I built it last week, I did not see the copy link  (I was also using it on mobile network and not wifi) 

Seems to be in order now!

**Reply by skibitsky** [reply]
The role of the Explorer backend is limited to providing data about wallets and some icons. The `Copy Link` button and QR code may need some time to appear on a slow network. They may also not appear if connectivity is lost or a network anomaly occurs.

If you ever again notice them not appearing, please open an issue; it could also be a bug.
